src/core/net/srp_client.cpp

```diff
@@ -1500,51 +1503,86 @@ void Client::EnableAutoStartMode(AutoStartCallback aCallback, void *aContext)
 
 void Client::ProcessAutoStart(void)
 {
-    uint16_t                                numServers = 0;
-    NetworkData::Service::SrpServer::Info   selectedServer;
-    NetworkData::Service::SrpServer::Info   server;
-    NetworkData::Service::Manager::Iterator iterator;
+    Ip6::SockAddr                             serverSockAddr;
+    bool                                      serverIsAnycast = false;
+    NetworkData::Service::DnsSrpAnycast::Info anycastInfo;
 
     VerifyOrExit(mAutoStartModeEnabled);
 
-    // If the client is not running we check if there is any SRP sever
-    // info in Network Data and select one randomly and then start the
-    // client. If the client is already running with a server that was
-    // selected by the auto-start feature, we verify that the selected
-    // server is still present in the Network Data.
+    serverSockAddr.Clear();
+
+    // If the SRP client is not running and auto start mode is
+    // enabled, we check if we can find any SRP server info in the
+    // Thread Network Data. If it is already running and the server
+    // was chosen by the auto-start feature, then we ensure that the
+    // selected server is still present in the Network Data.
+    //
+    // Two types of "DNS/SRP Service" entries can be present in
+    // Network Data, "DNS/SRP Service Anycast Address" model and
+    // "DNS/SRP Service Unicast" model. The Anycast entries are
+    // preferred over the Unicast entries.
 
     VerifyOrExit(!IsRunning() || mAutoStartDidSelectServer);
 
-    while (Get<NetworkData::Service::Manager>().GetNextSrpServerInfo(iterator, server) == kErrorNone)
+    if (Get<NetworkData::Service::Manager>().FindPreferredDnsSrpAnycastInfo(anycastInfo) == kErrorNone)
     {
-        numServers++;
-
-        // Choose a server randomly (with uniform distribution) from
-        // the list of servers. As we iterate through server entries,
-        // with probability `1/numServers`, we choose to switch the
-        // current selected server with the new entry. This approach
-        // results in a uniform/same probability of selection among
-        // all server entries.
-
-        if ((numServers == 1) || (Random::NonCrypto::GetUint16InRange(0, numServers) == 0))
+        if (IsRunning() && mAutoStartDidSelectServer && mAutoStartIsUsingAnycastAddress)
```

Comment:
```suggestion
        if (IsRunning() && mAutoStartIsUsingAnycastAddress)
```
`IsRunning()` already implies `mAutoStartDidSelectServer`

Reply:
Not necessarily. 

The SRP clinet an be started by a call to `Start(aAddress)` in which case the address is given by user (and not selected by auto start feature).